packages/block-library/src/navigation/view.js

```diff
@@ -73,60 +73,77 @@ store( {
 	selectors: {
 		core: {
 			navigation: {
-				roleAttribute: ( { context, selectors } ) =>
-					context.core.navigation.overlay &&
-					selectors.core.navigation.isMenuOpen( { context } )
+				roleAttribute: ( store ) => {
+					const { context, selectors } = store;
+					return context.core.navigation.type === 'overlay' &&
+						selectors.core.navigation.isMenuOpen( store )
 						? 'dialog'
-						: '',
+						: '';
+				},
 				isMenuOpen: ( { context } ) =>
```

Comment:
Shouldn't we use `store` here for consistency?

Reply:
Oh, I'm using `store` only when the function is using at least one selector 🙂